src/app/components/query-explainer/query-explainer.component.ts

```diff
@@ -0,0 +1,144 @@
+import { ChangeDetectionStrategy, Component, inject, OnInit } from "@angular/core";
+import AppValues from "src/app/common/app.values";
+import { Clipboard } from "@angular/cdk/clipboard";
+import { QueryExplainerService } from "./query-explainer.service";
+import { BaseComponent } from "src/app/common/base.component";
+import { MaybeUndefined } from "src/app/common/app.types";
+import ProjectLinks from "src/app/project-links";
+import { catchError, map, Observable, of, switchMap, tap } from "rxjs";
+import {
+    QueryExplainerDatasetsType,
+    QueryExplainerResponse,
+    VerifyQueryDatasetBlockNotFoundError,
+    VerifyQueryDatasetNotFoundError,
+    VerifyQueryError,
+    VerifyQueryKindError,
+    VerifyQueryResponse,
+} from "./query-explainer.types";
+import { HttpErrorResponse } from "@angular/common/http";
+import { BlockService } from "src/app/dataset-block/metadata-block/block.service";
+import { changeCopyIcon } from "src/app/common/app.helpers";
+
+@Component({
+    selector: "app-query-explainer",
+    templateUrl: "./query-explainer.component.html",
+    styleUrls: ["./query-explainer.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class QueryExplainerComponent extends BaseComponent implements OnInit {
+    private clipboard = inject(Clipboard);
+    private queryExplainerService = inject(QueryExplainerService);
+    private blockService = inject(BlockService);
+    public readonly DATE_FORMAT = AppValues.DISPLAY_FLOW_DATE_FORMAT;
+    public readonly VerifyQueryKindError: typeof VerifyQueryKindError = VerifyQueryKindError;
+
+    public sqlQueryExplainerResponse: QueryExplainerResponse;
```

Comment:
I still think we can avoid this variable and use observables only.

Reply:
Fixed